source/source_test.go

```diff
@@ -16,15 +16,450 @@ package source
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
+	"fmt"
+	"os"
+	"strings"
 	"testing"
+	"time"
 
+	"cloud.google.com/go/storage"
 	"github.com/conduitio/conduit-connector-google-cloudstorage/config"
+	sourceConfig "github.com/conduitio/conduit-connector-google-cloudstorage/source/config"
+	"github.com/conduitio/conduit-connector-google-cloudstorage/source/position"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
 )
 
+type Object struct {
+	key     string
+	content string
+}
+
+const (
+	projectID = "projectID"
+)
+
+func TestSource_SuccessfulSnapshot(t *testing.T) {
+	client, cfg := prepareIntegrationTest(t)
+
+	ctx := context.Background()
+	testBucket := cfg[config.ConfigKeyGCSBucket]
+	source := &Source{}
```

Comment:
Initially did the same but because of the linting issue reverted it back. Let me use the anonymous defer function to handle the source.Teardown() error for avoiding the linting issues.

Reply:
Changes are done :)
Approve it if there are no improvements so i would merge it.

Later i will create a another issue to update the go test GitHub action to run the integration tests with the help of the configured GitHub secured variables.

Reply:
Unless it's much work to add the GH secrets, it would be good if we add them before merging this PR. Seeing it green will make us all feel good about it.: )

Reply:
Completed Waiting for Approval to merge

Reply:
For some reason, I can't resolve this conversation. :+1: from me.

Reply:
May i know is anything left here ??
Because currently iam using the defer block for source teardown and even add the GH secrets to run the integration tests via github actions

Reply:
May i know is anything left here ??
Because currently we are using the defer block for source tear down and even add the GitHub secrets to run the integration tests via GitHub actions

Reply:
I think we are good here. :+1: